### PR TITLE
LsiModel.docs_processed attribute

### PR DESCRIPTION
**Pull request by hobson**
unittests in test_lsimodel.py and test_corpora.py
works for test MmCorpus in unittests
also works for large custom corpora in production app

### CODE REVIEW

File: gensim/models/lsimodel.py [hunk 1]
```diff
@@ -395,6 +396,7 @@ def add_documents(self, corpus, chunksize=None, decay=None):
                 if self.dispatcher:
                     logger.info("reached the end of input; now waiting for all remaining jobs to finish")
                     self.projection = self.dispatcher.getstate()
+                self.docs_processed += len(corpus) if hasattr(corpus, '__len__') else doc_no
```
Comment (tmylk) [inline gensim/models/lsimodel.py]: can we always use doc_no?

Comment (hobson) [inline gensim/models/lsimodel.py]: Yea, I guess so, for this line.

File: gensim/models/lsimodel.py [hunk 2]
```diff
@@ -361,6 +361,7 @@ def add_documents(self, corpus, chunksize=None, decay=None):
                     num_terms=self.num_terms, chunksize=chunksize,
                     extra_dims=self.extra_samples, power_iters=self.power_iters)
                 self.projection.merge(update, decay=decay)
+                self.docs_processed += len(corpus) if hasattr(corpus, '__len__') else 0
```
Comment (piskvorky) [inline gensim/models/lsimodel.py]: you can pass a default value to `getattr()`

File: gensim/test/test_corpora.py [hunk 3]
```diff
@@ -16,14 +16,17 @@
 
 import numpy
 
-from gensim.utils import to_unicode, smart_extension
+from gensim.utils import to_unicode  # , smart_extension
```
Comment (piskvorky) [inline gensim/test/test_corpora.py]: Remove import if no longer needed.

**Comment by piskvorky** [comment]
Nice! These type of fixes are very valuable.

What was your motivation for this PR @hobson ?

**Comment by hobson** [comment]
`docs_processed += getattr(corpus, '__len__', int)()`

Seems less explicit, less pythonic to me, but happy to do it if you like.


I was debugging my training on an iterable QuerySetCorpus class for a
variable-length database query result. The existing attribute was always
zero, confusing. Need to know exactly how many records trained my model.

File: gensim/test/test_corpora.py (hunk 3 above)
Comment (hobson) [inline gensim/test/test_corpora.py]: Done.

Side Note: There are many more unused imports throughout gensim. They can be dangerous to remove, though, for someone like me unfamiliar with the internals of those packages being imported. For example `import seaborn` has side-effects, and obviously `from future import division` does too.

**Comment by tmylk** [comment]
Agree. 
Thanks for the PR!

**Comment by tmylk** [comment]
Just a line in the CHANGELOG and then will merge.

**Pull request merged**